aiida_quantumespresso/parsers/pp.py

```diff
@@ -0,0 +1,383 @@
+                data.append(float(split_line[2]))
+
+        coords_units = 'bohr'
+        data_units = self.units_dict[self.output_parameters['plot_num']]
+        arraydata = orm.ArrayData()
+        arraydata.set_array('xy_coordinates', np.array(coords))
+        arraydata.set_array('data', np.array(data))
+        arraydata.set_array('xy_coordinates_units', np.array(coords_units))
+        arraydata.set_array('data_units', np.array(data_units))
+
+        return arraydata
+
+
+    def parse_gaussian(self, data_file_str):
+        """
+        Parse Gaussian Cube formatted output
+
+        :param data_file_str: the data file read in as a single string
+        """
+
+        lines = data_file_str.splitlines()
+
+        atoms_line = lines[2].split()
+        atoms = int(atoms_line[0]) # The number of atoms listed in the file
+        header = lines[:6+atoms]  # The header of the file: comments, the voxel, and the number of atoms and datapoints
+        data_lines = lines[6+atoms:] # The actual data: atoms and volumetric data
+
+        # Parse the declared dimensions of the volumetric data
+        x_line = header[3].split()
+        xdim = int(x_line[0])
+        y_line = header[4].split()
+        ydim = int(y_line[0])
+        z_line = header[5].split()
+        zdim = int(z_line[0])
+
+        #Get the vectors describing the basis voxel
+        voxel_array = np.array(
+            [[x_line[1], x_line[2], x_line[3]],
+            [y_line[1], y_line[2], y_line[3]],
+            [z_line[1], z_line[2], z_line[3]]],
+            dtype=np.float64
+        )
+
+        # Get the volumetric data
+        data_array = np.zeros((xdim, ydim, zdim))
+
+        # The data is organised into columns with a fixed number of values
+        # Gather up all the data points into one list for easier unpacking
+        datalist = []
+        for line in data_lines:
+            for i in range(0, len(line), 13):
+                data_point = line[i:i+13].strip()
+                if data_point != '':
+                    datalist.append(float(data_point))
+
+        # Unpack the list and repack as a 3D array
+        # Note the unusual indexing: cube files run over the z index first, then y and x.
+        # E.g. The first volumetric data point is x,y,z = (0,0,0) and the second is (0,0,1)
+        for i in range(0, xdim):
+            for j in range(0, ydim):
+                for k in range(0, zdim):
+                    data_array[i, j, k] = (datalist[(i*ydim*zdim)+(j*zdim)+k])
+
+        coordinates_units = 'bohr'
+        data_units = self.units_dict[self.output_parameters['plot_num']]
+
+        arraydata = orm.ArrayData()
+        arraydata.set_array('voxel', voxel_array)
+        arraydata.set_array('data', data_array)
+        arraydata.set_array('coordinates_units', np.array(coordinates_units))
+        arraydata.set_array('data_units', np.array(data_units))
+
+        return arraydata
+
+
+    def emit_logs(self, *args):
```

Comment:
This is now implemented on the base class `aiida_quantumespresso.parsers.base.Parser` and so you can remove it here

Reply:
Dropped.